Makefile.targets.mk

```diff
@@ -12,24 +59,43 @@ REV=$(shell git rev-parse --short HEAD)
 # Tag is the tag to use for build and push image targets.
 TAG ?= $(REV)
 
-.PHONY: build
-build:  ## Build the envoy-gateway binary
-	@CGO_ENABLED=0 go build -a -o ./bin/${GOOS}/${GOARCH}/ github.com/envoyproxy/gateway/cmd/envoy-gateway
+# Build the target binary in target platform.
+# The pattern of build.% is `build.{Platform}.{Command}`.
+# If we want to build envoy-gateway in linux amd64 platform, 
+# just execute make build.linux_amd64.envoy-gateway.
+.PHONY: build.%
+build.%:
+	$(eval COMMAND := $(word 2,$(subst ., ,$*)))
+	$(eval PLATFORM := $(word 1,$(subst ., ,$*)))
+	$(eval OS := $(word 1,$(subst _, ,$(PLATFORM))))
+	$(eval ARCH := $(word 2,$(subst _, ,$(PLATFORM))))
+	@echo "===========> Building binary $(COMMAND) $(VERSION) for $(OS) $(ARCH)"
+	@mkdir -p $(OUTPUT_DIR)/$(OS)/$(ARCH)/
+	@CGO_ENABLED=0 GOOS=$(OS) GOARCH=$(ARCH) go build -o $(OUTPUT_DIR)/$(OS)/$(ARCH)/$(COMMAND) $(ROOT_PACKAGE)/cmd/$(COMMAND)
 
-build-linux-amd64:
-	@GOOS=linux GOARCH=amd64 $(MAKE) build
+# Build the envoy-gateway binaries in the hosted platforms.
```

Comment:
do these comments show up in `make help`

Reply:
sure